connector/protobuf/src/main/scala/org/apache/spark/sql/protobuf/functions.scala

```diff
@@ -148,8 +212,38 @@ object functions {
     messageName: String,
     descFilePath: String,
     options: java.util.Map[String, String]): Column = {
+    val fileContent = ProtobufUtils.readDescriptorFileContent(descFilePath)
```

Comment:
Can't we change CatalystDataToProtobuf and ProtobufDataToCatalyst to accept descFilePath or fileDescriptorSetBytes as optional parameters instead of passing them in functions? Perhaps we could throw an error if both are not set?  This is just a thought.